template/content/test/child-process.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+var console = require('console');
+var path = require('path');
+var spawn = require('child_process').spawn;
+var setTimeout = require('timers').setTimeout;
+
+var TestCluster = require('./lib/test-cluster.js');
+
+var serverFile = path.join(
+    __dirname, '..', 'server.js'
+);
+
+TestCluster.test('spin up server', {
+    appCount: 0
+}, function t(cluster, assert) {
+    var args = [
+        serverFile,
+        '--port', '0'
+    ];
+
+    var hostPortList = cluster.hyperbahnCluster.hostPortList;
```

Comment:
It uses TestCluster with appCount: 0 to get `cluster.hyperbahnCluster` and `cluster.client`

This means it tells the child process to advertise with hyperbahn and then uses `cluster.client` to talk to the child process through hyperbahn.

The `TestCluster` does three things

 - spins up a hyperbahn ring of size 2
 - creates your `test-client.js` which talks to hyperbahn
 - spins up an app ring of size N.